crates/state/src/da_blob.rs

```diff
@@ -115,3 +118,30 @@ impl BlobIntent {
         }
     }
 }
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, Arbitrary, BorshDeserialize, BorshSerialize)]
+pub struct BlobCommitment(pub Buf32);
+
+impl BlobCommitment {
+    pub fn new(commitment: &Buf32) -> Self {
+        Self(*commitment)
+    }
+
+    pub fn into_inner(&self) -> Buf32 {
+        self.0
+    }
+
+    pub fn from_payload(payload: &[InscriptionBlob]) -> Self {
+        let commitment = payload.iter().fold(Buf32::zero(), |acc, blob| {
+            hash::raw(&[acc.0, hash::raw(blob.data()).0].concat())
+        });
+        Self(commitment)
+    }
+
+    pub fn verify_against_payload(&self, payload: &[InscriptionBlob]) -> bool {
+        let commitment = payload.iter().fold(Buf32::zero(), |acc, blob| {
+            hash::raw(&[acc.0, hash::raw(blob.data()).0].concat())
+        });
+        commitment == self.0
+    }
```

Comment:
We had a short discussion on whether to use merkle trees or not. Do we actually need to prove existence any entry? I couldn't think of cases when we would need to prove the entries.

Reply:
We don't.  They get tracked in the chain with the granularity of blobs so we should recognize them in terms of blobs.  This chained commitment scheme would break since we wouldn't know what we should be looking for in the L1 proofs.